controllers/productController.ts

```diff
@@ -14,28 +15,256 @@ const getAllProducts = async (req: Request, res: Response) => {
-  return;
+  const brandId = req.params.id;
+
+  db.Product.findAll({ where: { brandId } })
+    .then((products: any) => {
+      if (!products || products.length === 0) {
+        res
+          .status(404)
+          .json({ error: "No products found for the given brand ID" });
+        return;
+      }
+      res.json({ products });
+    })
+    .catch((error: Error) => {
+      // tslint:disable-next-line:no-console
+      console.error("Error finding products by brand ID:", error);
+      res.status(500).json({ error: "Internal server error" });
+    });
 };
 
 // This method creates a new product
 const createNewProduct = async (req: Request, res: Response) => {
   return;
 };
+const getNewArrivals = (req: Request, res: Response) => {
+  const threeMonthsAgo = new Date();
+  threeMonthsAgo.setMonth(threeMonthsAgo.getMonth() - 3);
 
+  db.Product.findAll({
+    where: {
+      createdAt: {
+        [Op.gt]: threeMonthsAgo,
+      },
+    },
+    attributes: [
+      "id",
+      "title",
+      "subtitle",
+      "description",
+      "price",
+      "quantity",
+      "discountpercentage",
+      "createdAt",
+      "updatedAt",
+      "brandId",
+      "cartId",
+      "categoryId",
+      "wishlistId",
+    ],
+  })
+    .then((products: any) => {
+      if (products.length === 0) {
+        res.status(404).json({ message: "No new arrival products found." });
+        return;
+      }
+      res.json({ products });
+    })
+    .catch((error: Error) => {
+      console.error("Error retrieving new arrival products:", error);
+      res.status(500).json({ error: "Database error", details: error.message });
+    });
+};
+const getHandPickedProducts = (req: Request, res: Response): void => {
+  db.Product.findAll({
+    include: [
+      {
+        model: db.Review,
+        as: "Reviews",
+        attributes: [],
+      },
+    ],
+    where: {
+      price: {
+        [Op.lt]: 100,
+      },
+    },
+    group: ["Product.id"],
+    having: literal("AVG(`Reviews`.`rating`) > 4.5"),
+    attributes: [
+      "id",
+      "title",
+      "subtitle",
+      "price",
+      "quantity",
+      "discountpercentage",
+      [fn("AVG", col("`Reviews`.`rating`")), "averageRating"],
+    ],
+    order: [[fn("AVG", col("`Reviews`.`rating`")), "DESC"]],
+    raw: true,
+  })
+    .then((products: any) => {
+      if (products.length === 0) {
+        res.status(404).json({ message: "No handpicked products found." });
+        return;
+      }
+      res.json({ products });
+    })
+    .catch((error: Error) => {
+      console.error("Error retrieving handpicked products:", error);
+      res.status(500).json({ error: "Database error", details: error.message });
+    });
+};
+const getLimitedEditionProducts = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const products = await db.Product.findAll({
+      where: {
+        quantity: {
+          [db.Sequelize.Op.lt]: 20,
+        },
+      },
+      attributes: [
```

Comment:
You don't need to add attributes if you list them all. However, I think we have to exclude some like updatedAt, you can modify it as:

attributes: { exclude: ['updatedAt'] },